### ISSUE TITLE
Calling savefig() changes the return value of ax.get_xticklabels()

### ISSUE THREAD

**Issue by levitsky** [matplotlib/matplotlib#6800]
`matplotlib` 1.5.1 on Linux, installed from binary package, both on Python 3.5.2 and 2.7.12.

Here is the code:

``` python
import pylab
f, axes = pylab.subplots(2, sharex=True, sharey=True)
for ax in axes:
    ax.plot(range(100))
for ax in axes:
    print([x.get_text() for x in ax.get_xticklabels()])
print('----')
pylab.savefig('/tmp/tempfig.png')
for ax in axes:
    print([x.get_text() for x in ax.get_xticklabels()])
```

The output is:

```
['', '', '', '', '', '']
['', '', '', '', '', '']
----
['0', '20', '40', '60', '80', '100']
['0', '20', '40', '60', '80', '100']
```

If I comment out the call to `savefig()`, I get

```
['', '', '', '', '', '']
['', '', '', '', '', '']
----
['', '', '', '', '', '']
['', '', '', '', '', '']
```

The tick labels (which I am trying to get) are seen in the figure:

![image](https://cloud.githubusercontent.com/assets/3358268/16960270/5d0b5976-4df1-11e6-89e6-a86a55b9f0eb.png)

Is this expected?

**Comment by WeatherGod** [comment]
Yes, sort of. This is one of the warts in matplotlib's design. Several
things are deferred until draw time. Axes ticks are one of them. There are
similar problems with facecolors of some scalar mappables that are
initialized to blue to start but won't get their final color until draw
time.

It isn't a bug, per se, but it is a wart. We really should make it so that
calls like get_xticklabels() forces the artist to not defer any longer and
compute their relevant values.

**Comment by story645** [comment]
This is kinda a dupe of #6638 so wondering if at least for now it should at the least be a doc patch? (Where though...)

**Comment by levitsky** [comment]
This is indeed a dupe, sorry for not searching thoroughly enough. Feel free to close it as such or do whatever you think is better.
At least I get the idea now, thank you.

**Comment by story645** [comment]
It's fine! ('specially since I created the other issue...). I'm just wondering out loud about where something can be slotted in to alleviate some of this confusion.

**Comment by tacaswell** [comment]
A better question is _why_ are you trying to do this?

**Comment by levitsky** [comment]
That's because I don't know the API so well.
I was trying to re-set the tick labels just to change their font size.
I have already figured out that I can use `setp()` for that.

**Comment by story645** [comment]
Oh, @tacaswell what about a warning on the get_methods if you try and call 'em before the figure is drawn? (or is that way too much overhead...)

**Comment by WeatherGod** [comment]
The problem is that it is only some methods that are problematic, and I
don't think anybody has a clear idea which ones they are.

**Comment by QuLogic** [comment]
This was fixed by #23170.
